### PR TITLE
Ports TG Hallucinations

### PR DESCRIPTION
**Pull request by psykzz**
## About The Pull Request
Ports TG Hallucinations
Adding TGMC specific hallucinations that make sense

## Why It's Good For The Game

Fun times - https://streamable.com/bduey
(Note the video is ramped up to show more)

## Changelog
:cl:
add: new hallucinations
/:cl:

### CODE REVIEW

**Comment by Rohesie** [comment]
Blessed content!

File: code/_onclick/ventcrawl.dm [hunk 1]
```diff
@@ -90,7 +90,7 @@ GLOBAL_LIST_INIT(ventcrawl_machinery, typecacheof(list(
 
 			visible_message("<span class='notice'>[src] scrambles into the ventilation ducts!</span>","<span class='notice'>You climb into the ventilation ducts.</span>")
 			if(!isxenohunter(src)) //Hunters silently enter/exit vents.
-				pick(playsound(src, 'sound/effects/alien_ventpass1.ogg', 35, 1), playsound(src, 'sound/effects/alien_ventpass2.ogg', 35, 1))
+				playsound(src, get_sfx("alien_ventpass"), 35, 1)
```
Comment (Rohesie) [inline code/_onclick/ventcrawl.dm]:
```suggestion
				playsound(src, get_sfx("alien_ventpass"), 35, TRUE)
```

File: code/modules/atmospherics/machinery/atmosmachinery.dm [hunk 2]
```diff
@@ -261,7 +261,7 @@
 	user.visible_message("<span class='warning'>[user] climbs out of [src].</span>", \
 	"<span class='notice'>You climb out of [src].</span>")
 	if(!isxenohunter(user) )
-		pick(playsound(user, 'sound/effects/alien_ventpass1.ogg', 35, 1), playsound(user, 'sound/effects/alien_ventpass2.ogg', 35, 1))
+		playsound(src, get_sfx("alien_ventpass"), 35, 1)
```
Comment (Rohesie) [inline code/modules/atmospherics/machinery/atmosmachinery.dm]:
```suggestion
		playsound(src, get_sfx("alien_ventpass"), 35, TRUE)
```

File: code/modules/flufftext/Hallucination.dm [hunk 3]
```diff
@@ -1,420 +1,292 @@
...
+	))
+
+/mob/living/carbon
+	// halucination vars
+	var/image/halimage
+	var/image/halbody
+	var/obj/halitem
+	var/hal_screwyhud = SCREWYHUD_NONE
+	var/next_hallucination = 0
+	var/cpr_time = 1 //CPR cooldown.
```
Comment (Rohesie) [inline code/modules/flufftext/Hallucination.dm]: Could you place these variable on the carbon definition?

Comment (psykzz) [reply code/modules/flufftext/Hallucination.dm]: Yeah it was

File: code/__DEFINES/mobs.dm [hunk 4]
```diff
@@ -599,3 +599,11 @@ GLOBAL_LIST_INIT(xenoupgradetiers, list(XENO_UPGRADE_BASETYPE, XENO_UPGRADE_INVA
...
 
 #define HUMAN_CARRY_SLOWDOWN 0.35
+
+
+// =============================
+// Hallucinations - health hud screws for carbon mobs
+#define SCREWYHUD_NONE 0
+#define SCREWYHUD_CRIT 1
+#define SCREWYHUD_DEAD 2
+#define SCREWYHUD_HEALTHY 3
```
Comment (Rohesie) [inline code/__DEFINES/mobs.dm]:
```suggestion
#define SCREWYHUD_HEALTHY 3

```
Linter

**Comment by psykzz** [comment]
Good to merge

**Pull request merged**